packages/std/src/imports.rs

```diff
@@ -182,7 +182,12 @@ impl Api for ExternalApi {
         Ok(address.into())
     }
 
-    fn secp256k1_verify(&self, message_hash: &[u8], signature: &[u8], public_key: &[u8]) -> bool {
+    fn secp256k1_verify(
+        &self,
+        message_hash: &[u8],
+        signature: &[u8],
+        public_key: &[u8],
+    ) -> StdResult<bool> {
```

Comment:
Let's create something more specific here: `-> Result<bool, VerificationError>`, which is similar to CryptoError, but independent.

`enum VerificationError` gets all the known error cases and an `VerificationError::Unknown` for the rest. We then implement `impl From<VerificationError> for StdError` which uses the string conversion below.

This allows contract developers to use the error codes or forget about then (via `?`).